README.md

```diff
@@ -1,111 +1,108 @@
+This directory must contain the desired network state for all hosts in a <i>hostname</i>.yaml file format.
+Please refer to the official nmstate docs for [examples](https://nmstate.io/examples.html).
 
-For AMD64 / x86_64 based systems:
 ```shell
-$ curl -o nm-configurator -L https://github.com/suse-edge/nm-configurator/releases/latest/download/nm-configurator-amd64 
-$ chmod +x nm-configurator
+$ export RUST_LOG=info # set log level ("error" by default)
+$ ./nmc generate --config-dir config --output-dir _out
+[2023-11-08T21:18:12Z INFO  nmc::generate_conf] Generating config from "config/node1.yaml"...
+[2023-11-08T21:18:12Z INFO  nmc::generate_conf] Generating config from "config/node2.yaml"...
+[2023-11-08T21:18:12Z INFO  nmc::generate_conf] Generating config from "config/node3.yaml"...
+[2023-11-08T21:18:12Z INFO  nmc] Successfully generated and stored network config
 ```
 
-For ARM64 based systems:
+The output is the following:
+
 ```shell
-$ curl -o nm-configurator -L https://github.com/suse-edge/nm-configurator/releases/latest/download/nm-configurator-arm64 
-$ chmod +x nm-configurator
-```
+$ ls -R _out
+_out:
+host_config.yaml  node1  node2  node3
 
-### Manual method:
+_out/node1:
+bond0.nmconnection  eth0.nmconnection
 
-```shell
-$ git clone https://github.com/suse-edge/nm-configurator.git
-$ cd nm-configurator
-$ go build . # optionally specify GOOS and GOARCH flags if cross compiling
+_out/node2:
+eth0.nmconnection  eth1.nmconnection
+
+_out/node3:
+eth2.nmconnection
 ```
 
-## How to run it?
+There are separate directories for each host (identified by their input <i>hostname</i>.yaml).
+Each of these contains the configuration files for the desired network interfaces (e.g. `eth0`).
 
-Using an example configuration of three known nodes (with hostnames `node1.example.com`, `node2.example.com`
-and `node3.example.com` and their respective NetworkManager settings) and interface mapping defined in `host_config.yaml`:
-
-```text
-config
-├── node1.example.com
-│   ├── eth0.nmconnection
-│   └── eth1.nmconnection
-├── node2.example.com
-│   └── eth0.nmconnection
-├── node3.example.com
-│   ├── bond0.nmconnection
-│   └── eth1.nmconnection
-└── host_config.yaml
-```
+The `host_config.yaml` file on the root level maps the hosts to all of their preconfigured interfaces.
+This is necessary in order for NMC to identify which host it is running on when applying the network configurations later.
 
-```shell
-$ ./nm-configurator -config-dir=config -hosts-config-file=host_config.yaml
-INFO[2023-08-17T17:32:23+03:00] starting network manager configurator...
-INFO[2023-08-17T17:32:23+03:00] successfully identified host: node1.example.com
-INFO[2023-08-17T17:32:23+03:00] storing file /etc/NetworkManager/system-connections/eth0.nmconnection... 
-INFO[2023-08-17T17:32:23+03:00] storing file /etc/NetworkManager/system-connections/eth1.nmconnection... 
-INFO[2023-08-17T17:32:23+03:00] successfully configured network manager 
+```yaml
+- hostname: node1
```

Comment:
OK, so the `host_config.yaml` is here, but shouldn't it be defined/documented prior to the `generate` above?

Reply:
This is just an example of how it looks. Users shouldn't be creating or modifying it. Do you think specifying a label of some kind will make this more clear?